### PR TITLE
Replaced golang Sarama kafka interface with Confluent.

### PR DESCRIPTION
**Pull request by mpkelly-hpe**
### Summary and Scope

Replaced golang Sarama kafka interface with Confluent.
Sarama interface can be buggy and can miss messages, etc.
Confluent is more stable and mature.

No API changes were made; this change is 100% backward compatible.

DOES THIS CHANGE INVOLVE ANY SCHEME CHANGES?  N

REMINDER: HAVE YOU INCREMENTED VERSION NUMBERS? E.G., .spec, Chart.yaml Y

REMINDER 2: HAVE YOU UPDATED THE COPYRIGHT PER hpe GUIDELINES: © Copyright 2014-2020 Hewlett Packard Enterprise Development LP    ? Y

### Issues and Related PRs

* Resolves CASMHMS-5299

### Testing

Tested on:

* Local build system
* baldar

Was a fresh Install tested? N   Just a go lib, not a microservice.
Was an Upgrade tested?      N   " "
Was a Downgrade tested?     N   " "
Was a CT test run?          N   " "
If schema changes were part of this change, how were those handled in your upgrade/downgrade testing? N/A

Created a special version of HMNFD which contained the new hms-msgbus
library.  Verified that it was able to connect, send, and receive
messages on the Kafka bus.

Local build machine testing included stress testing, consumer
group testing, etc.  All worked as expected.

### Risks and Mitigations

Should be low risk, since the Confluent golang package is mature
and stable.

### CODE REVIEW

**Review by nieuwsma — COMMENTED** [review]
please document the -tags musl build args in a readme.md

**Review by nieuwsma — COMMENTED** [review]
should we pass in a logger? come back to this Andrew; this might make this a 2.0

File: msgbus.go [hunk 1]
```diff
@@ -167,30 +180,44 @@ func (mbus *MsgBusWriter_Kafka) Status() int {
...
+		err = mbus.kafkaProducer.Produce(&kafka.Message{
+			TopicPartition: kafka.TopicPartition{Topic: &mbus.kafkaTopic, Partition: kafka.PartitionAny},
+			Value:          []byte(msg),
+		}, deliveryChan)
+
+		e := <-deliveryChan
+		m := e.(*kafka.Message)
+
+		if m.TopicPartition.Error != nil {
+			log.Printf("Kafka message delivery failed: %v\n", m.TopicPartition.Error)
```
Comment (nieuwsma) [inline msgbus.go:208]: prepend with "ERROR"

Comment (nieuwsma) [reply msgbus.go:208]: also print some meta data like : topic.   We probably dont want to print the message as those could be big and sensitive.   Any other meta data we can expose or should expose?

**Review by nieuwsma — CHANGES_REQUESTED** [review]
Please add the global logger as we discussed.

**Review by mjendrysik-hpe — APPROVED** [review]

File: msgbus.go (hunk 1 above)
Comment (mpkelly-hpe) [reply msgbus.go]: Fixed.

**Review by nieuwsma — APPROVED** [review]

**Pull request merged**